sippy-ng/src/component_readiness/SavedViews.js

```diff
@@ -0,0 +1,248 @@
+        },
+      }
+
+      // Append the new view to the existing views.
+      const newViews = {
+        ...views,
+        [newViewName]: newViewConfig,
+      }
+      setViews(newViews)
+
+      // Update on save.
+      varsContext.saveViewsToLocal(newViews)
+      handleCloseSaveDialog()
+    }
+  }
+
+  const openDeleteDialog = (viewName) => {
+    setViewToDelete(viewName)
+    setDeleteDialogIsOpen(true)
+  }
+
+  const closeDeleteDialog = () => {
+    setDeleteDialogIsOpen(false)
+    setViewToDelete(null)
+  }
+
+  const userViewDelete = () => {
+    const newViews = { ...views }
+    delete newViews[viewToDelete]
+    setViews(newViews)
+
+    // Update on delete.
+    varsContext.saveViewsToLocal(newViews)
+
+    // User deleted their current view, so switch to Default.
+    if (viewToDelete === buttonName) {
+      setButtonName(Object.keys(newViews)[0])
+      setViewParam(Object.keys(newViews)[0])
+    }
+    closeDeleteDialog()
+  }
+
+  return (
+    <Fragment>
+      <Button
+        aria-controls="view-menu"
+        aria-haspopup="true"
+        startIcon={<ViewCarousel />}
+        color="primary"
+        onClick={handleClick}
+      >
+        <Tooltip title={views[buttonName].config.help}>
+          View: {buttonName}
+        </Tooltip>
+      </Button>
+      <Menu
+        id="view-menu"
+        anchorEl={anchor}
+        keepMounted
+        open={Boolean(anchor)}
+        onClose={handleClose}
+      >
+        {Object.entries(props.views).map(([e, v]) => (
+          <MenuItem
+            key={e}
+            style={{
+              fontWeight: props.view === e ? 'bold' : 'normal',
+            }}
+            onClick={() => {
+              props.applyView(e)
+              setButtonName(e)
+              handleClose()
+            }}
+          >
+            <Tooltip title={v.config.help}>{e}</Tooltip>
+            {/* only user created views can be deleted */}
+            {v.config.class && v.config.class == 'user' ? (
+              <IconButton
+                edge="end"
+                aria-label="delete"
+                onClick={(event) => {
+                  // Don't fire onclick.
+                  event.stopPropagation()
+                  openDeleteDialog(e)
+                }}
+              >
+                <Tooltip title="Delete View">
+                  <DeleteIcon />
+                </Tooltip>
+              </IconButton>
+            ) : (
+              ''
+            )}
+          </MenuItem>
+        ))}
+      </Menu>
+
+      <Button
+        color="primary"
+        startIcon={<Save />}
+        onClick={handleOpenSaveDialog}
+      >
+        Save Current View
+      </Button>
+
+      <Dialog open={saveDialogIsOpen} onClose={handleCloseSaveDialog}>
+        <DialogTitle>Save Current View</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            Name for new view (max {maxSavedViewLength} characters)
```

Comment:
Why is the max 15 characters?

Can you also add some text explaining this is in the local browser storage only.

It might be nice if there was some kind of share button that let the user save the view in their own storage, but not required.

Reply:
I originally set it to 15 thinking it would make the left sidebar too wide.  Turns out it renders/wraps ok for longer names so I made the max 100 characters.

For that last part, do you mean save the views to the user's disk?